### PR TITLE
Detect virt next

### PR DESCRIPTION
**Pull request by shibumi**
implemented virtualization detection with systemd.

### CODE REVIEW

File: powerlevel9k.zsh-theme [hunk 1]
```diff
@@ -547,6 +547,26 @@ prompt_history() {
...
+prompt_detect_virt() {
+  if ! command -v systemd-detect-virt;then
+    return
+  fi
+  local virt=$(systemd-detect-virt)
+  local color="yellow"
+  if [[ "$virt" == "none" ]]; then
+    if [[ "$(ls -di / | grep -o 2)" != "2" ]]; then
+      virt="chroot"
+      "$1_prompt_segment" "$0" "$2" "$color" "$virt" ""
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: Here is still a parameter missing, right? Like said in #357

File: powerlevel9k.zsh-theme [hunk 2]
```diff
@@ -547,6 +547,26 @@ prompt_history() {
...
+    else
+      ;
+    fi
+  else
+    "$1_prompt_segment" "$0" "$2" "$color" "$virt" ""
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: Here as well. ;)

File: powerlevel9k.zsh-theme (hunk 1 above)
Comment (shibumi) [inline powerlevel9k.zsh-theme]: why? the last field is for the icon. I have no icon for the virtualization, just text.

Comment (dritter) [inline powerlevel9k.zsh-theme]: `$1_prompt_segment` takes six argument, you just provide five. The first is the name of the function (`$0`). Second is the index (`$2`). Third is the background color (`$color`). Fourth is foreground color (`$virt` in your case; this is what looks odd to me).. After that the arguments seem shifted.

Comment (dritter) [inline powerlevel9k.zsh-theme]: Yes, but there you omitted the argument for the icon (a.k.a. visual identifier, as I call it) which is optional. In the `detect-virt` segment, you omit the argument for foreground color which is a mandatory one.
As arguments of a function are positional, omitting one is always the last one (except you pass an empty parameter to the function; in ZSH terms this is an empty string, in other programming languages this is usually `null`). So passing 5 arguments omits the icon, but you pass an empty string. On the other hand you pass `$virt` as foreground color which seems wrong to me, as this should be passed one position later as the content of the segment.

**Comment by shibumi** [comment]
@dritter should be fixed now.

**Comment by bhilburn** [comment]
Nice work, @shibumi!! What a great contribution.

And thanks for the review work, @dritter! As always, much appreciated.

**Pull request merged**